pkg/auth/authLogin.go

```diff
@@ -0,0 +1,175 @@
+// {
+//   "jwt": "<bearer-token-here>"
+// }
+func writeBearerTokenJsonFile(fileSystem files.FileSystem, galasaHome utils.GalasaHome, jwt string) error {
+    bearerTokenFilePath := filepath.Join(galasaHome.GetNativeFolderPath(), "bearer-token.json")
+
+    log.Printf("Writing bearer token to file '%s'", bearerTokenFilePath)
+    err := fileSystem.WriteTextFile(bearerTokenFilePath, jwt)
+
+    if err == nil {
+        log.Printf("Written bearer token to file '%s' OK", bearerTokenFilePath)
+    } else {
+        log.Printf("Failed to write bearer token file '%s'", bearerTokenFilePath)
+        err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_FAILED_TO_WRITE_FILE, bearerTokenFilePath, err.Error())
+    }
+    return err
+}
+
+type BearerTokenJson struct {
+    Jwt string `json:"jwt"`
+}
+
+// Gets the JWT from the bearer-token.json file if it exists, errors if the file does not exist
+func getBearerTokenFromTokenJsonFile(fileSystem files.FileSystem, galasaHome utils.GalasaHome) (string, error) {
+    var err error = nil
+    var bearerToken string = ""
+    var bearerTokenJsonContents string = ""
+
+    bearerTokenFilePath := filepath.Join(galasaHome.GetNativeFolderPath(), "bearer-token.json")
+
+    log.Printf("Retrieving bearer token from file '%s'", bearerTokenFilePath)
+    bearerTokenJsonContents, err = fileSystem.ReadTextFile(bearerTokenFilePath)
+    if err == nil {
+        var bearerTokenJson BearerTokenJson
+        err = json.Unmarshal([]byte(bearerTokenJsonContents), &bearerTokenJson)
+        if err == nil {
+            bearerToken = bearerTokenJson.Jwt
+            log.Printf("Retrieved bearer token from file '%s' OK", bearerTokenFilePath)
+        }
+    }
+
+    if err != nil {
+        log.Printf("Could not retrieve bearer token from file '%s'", bearerTokenFilePath)
+        err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_RETRIEVING_BEARER_TOKEN_FROM_FILE, bearerTokenFilePath, err.Error())
+    }
+    return bearerToken, err
+}
+
+// Gets a new authenticated API client, attempting to log in if a bearer token file does not exist
+func GetAuthenticatedAPIClient(apiServerUrl string, fileSystem files.FileSystem, galasaHome utils.GalasaHome) *galasaapi.APIClient {
+    bearerToken, err := getBearerTokenFromTokenJsonFile(fileSystem, galasaHome)
+    if err != nil {
+        // Attempt to log in
+        log.Printf("Logging in to the Galasa Ecosystem at '%s'", apiServerUrl)
+        err = Login(apiServerUrl, fileSystem, galasaHome)
+        if err == nil {
+            log.Printf("Logged in to the Galasa Ecosystem at '%s' OK", apiServerUrl)
+            bearerToken, err = getBearerTokenFromTokenJsonFile(fileSystem, galasaHome)
+        }
+    }
+
+    var apiClient *galasaapi.APIClient
+    if err == nil {
+        apiClient = api.InitialiseAuthenticatedAPI(apiServerUrl, bearerToken)
+    } else {
+        // Continue without a bearer token
```

Comment:
Maybe a note saying this is temporary code ? And why it's here ? and when it can be removed ?

Reply:
Addressed in https://github.com/galasa-dev/cli/pull/151/commits/a3679e637189ed9792675899842f9067b570bf06